app/jobs/autotitle_conversation_job.rb

```diff
@@ -31,6 +31,12 @@ def generate_title_for(text)
         response_format: { type: "json_object" }  # this causes problems for Groq even though it's supported: https://console.groq.com/docs/api-reference#chat-create
       )
       return JSON.parse(response)["topic"]
+    elsif ai_backend.class == AIBackend::Gemini
+      response = ai_backend.get_oneoff_message(
+        system_message,
+        [text]
+      )
+      return response.split(":")[1].split("}")[0].gsub(/"/, "").strip
```

Comment:
Rather than parsing the JSON response manually, it looks like Gemini supports forced JSON responses:
https://github.com/gbaptista/gemini-ai#json-format-responses

Then you can do the same JSON.parse() as the other examples. Before I implemented the JSON mode for the other LLMs this method would randomly break and conversations would stop autotitling.